jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -306,25 +304,17 @@ def clean(self):
         self.display_uncertainties = False
         self.display_mask = False
 
-        marks = self.figure.marks[:]
-
         # Remove extra traces, in case they exist.
-        self._clean_error(marks)
-        self._clean_mask(marks)
-
-        self.figure.marks = marks
+        self._clean_error()
+        self._clean_mask()
 
-    def _clean_mask(self, marks):
-        if hasattr(self, 'mask_line_mark') and self.mask_line_mark is not None:
-            marks.remove(self.mask_line_mark)
-            self.mask_line_mark = None
-            self.mask_trace_pointer = None
+    def _clean_mask(self):
+        fig = self.figure
+        fig.marks = [x for x in fig.marks if not isinstance(x, ScatterMask)]
```

Comment:
should we just hide the marks instead of deleting them?  (I'm honestly not sure the performance implications and this isn't something that will be toggled repeatedly, so maybe its not worth it)

Reply:
I actually didn't know we could hide marks instead of deleting them...I'll look into that!

Reply:
I _think_ every mark has a `.visible` property that can be toggled, iirc.